usecase/src/main/java/org/mobilitydata/gtfsvalidator/usecase/ValidateMinTransferTimeValue.java

```diff
@@ -0,0 +1,88 @@
+ * - range min: 0 sec
+ * - range max: 86400 sec (24 hours)
+ *
+ * This use case is triggered after defining the content of the {@code GtfsDataRepository} provided in the constructor.
+ * At this step it it assumed that all values of `min_transfer_time` are positive integers.
+ */
+public class ValidateMinTransferTimeValue {
+    private final GtfsDataRepository dataRepo;
+    private final ValidationResultRepository resultRepo;
+    private final ExecParamRepository execParamRepo;
+    private final Logger logger;
+
+    /**
+     * @param dataRepo        a repository storing the data of a GTFS dataset
+     * @param resultRepo      a repository storing information about the validation process
+     * @param execParamRepo   a repository storing the execution parameters
+     * @param logger          a logger displaying information about the validation process
+     */
+    public ValidateMinTransferTimeValue(final GtfsDataRepository dataRepo,
+                                        final ValidationResultRepository resultRepo,
+                                        final ExecParamRepository execParamRepo,
+                                        final Logger logger) {
+        this.dataRepo = dataRepo;
+        this.resultRepo = resultRepo;
+        this.execParamRepo = execParamRepo;
+        this.logger = logger;
+    }
+
+    /**
+     * Use case execution method: checks if every record of file `transfers.txt` having a non null value for field
+     * min_transfer_time are within a certain range. This range is specified by the user of the software by execution
+     * parameters that are passed as command line arguments or defined within file `execution-parameter.json`.
+     * If this range is not specified, the use case will consider default values for said thresholds:
+     * - range min: 0 sec
+     * - range max: 86400 sec (24 hours)
+     */
+    public void execute() {
+        logger.info("Validating rule 'W009 - `min_transfer_time` is outside allowed range" + System.lineSeparator());
+        final int minTransferTimeRangeMin =
+                Integer.parseInt(
+                        execParamRepo.getExecParamValue(ExecParamRepository.TRANSFER_MIN_TRANSFER_TIME_RANGE_MIN));
+        final int minTransferTimeRangeMax =
+                Integer.parseInt(
+                        execParamRepo.getExecParamValue(ExecParamRepository.TRANSFER_MIN_TRANSFER_TIME_RANGE_MAX));
+        dataRepo.getTransferAll()
+                .forEach((fromStopId, transferCollection) -> transferCollection.forEach((toStopId, transfer) ->  {
+                    if (transfer.getMinTransferTime() != null &&
+                            (transfer.getMinTransferTime() < minTransferTimeRangeMin ||
+                                    transfer.getMinTransferTime() > minTransferTimeRangeMax)) {
+                        resultRepo.addNotice(
+                                new SuspiciousMinTransferTimeNotice(minTransferTimeRangeMin, minTransferTimeRangeMax,
```

Comment:
Wondering if it wouldn't be easier to just use the `IntegerValueOutOfRangeNotice` here?

Reply:
I would go with a custom notice as you did since the error is that `min_transfer_time` isn't right.

Reply:
I agree with using a custom notice. Looking at other uses of `IntegerValueOutOfRangeNotice`, they are clear errors (e.g., value less than 0), where as this is a warning based on a heuristic, where there could be valid values outside the chosen range.